core/scale/scale_decoder_stream.cpp

```diff
@@ -5,4 +5,143 @@
 
 #include "scale/scale_decoder_stream.hpp"
 
-namespace kagome::scale {}
+#include "macro/unreachable.hpp"
+#include "scale/outcome_throw.hpp"
+#include "scale/scale_error.hpp"
+#include "scale/types.hpp"
+
+namespace kagome::scale {
+  namespace {
+    CompactInteger decodeCompactInteger(ScaleDecoderStream &stream) {
+      auto first_byte = stream.nextByte();
+
+      const uint8_t flag = (first_byte)&0b00000011u;
```

Comment:
```suggestion
      const uint8_t kFlag = (first_byte)&0b00000011u;
```

Reply:
this const is not global, don't think it matters.